File: Logic Building Problems/Basic Problems/Sum of GP/method2.js
// Function to calculate sum of geometric series
function sumOfGP(a, r, n) {
	// calculating and storing sum
	return (a * (1 - Math.pow(r, n))) / (1 - r);
}

// Driver code
function main() {
	let a = 2; // first term
	let r = 2; // common ratio
	let n = 15; // number of terms

	console.log(sumOfGP(a, r, n));
}

// Run the main function
main();

File: Logic Building Problems/Basic Problems/Sum of GP/method2.java
// An Efficient solution to solve sum of 
// geometric series.
import java.math.*;

class Main{
	
	// function to calculate sum of 
	// geometric series
	static float sumOfGP(float a, float r, int n)
	{
		// calculating and storing sum
		return (a * (1 - (int)(Math.pow(r, n)))) / 
											(1 - r);
	}

	// driver code
	public static void main(String args[])
	{
		float a = 2; // first term
		float r = 2; // common ratio
		int n = 15; // number of terms

		System.out.println((int)(sumOfGP(a, r, n)));
	}
}

